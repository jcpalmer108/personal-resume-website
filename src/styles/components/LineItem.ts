import styled from "styled-components";
import { desktopBreakpoint, silver_polish } from "../variables";

export const Icon = styled.img`
  height: 20px;
`;

export const Info = styled.div`
  font-family: Gilroy-Regular;
  font-size: 18px;

  @media (min-width: ${desktopBreakpoint}px) {
    font-size: 16px;
  }
`;

export const Line = styled.div`
  display: flex;
  column-gap: 15px;
  padding: 7px 50px;

  &:last-child {
    padding-bottom: calc(35px - 7px);
  }
`;

export const Subtitle = styled.div`
  font-family: Gilroy-Regular;
  font-size: 11px;
  color: ${silver_polish};
  padding-top: 3px;
`;

export const SkillInfo = styled.div`
  display: flex;
  flex-direction: column;
`;
