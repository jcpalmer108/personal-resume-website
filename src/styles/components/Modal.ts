import styled from "styled-components"
import { desktopBreakpoint, jet, off_black, silver_polish, tabletBreakpoint } from "../variables";

export const BackgroundOverlay = styled.div`
  position: fixed;
  top: 0px;
  background: black;
  height: 100%;
  width: 100%;
  z-index: 998;
  opacity: 0.8;
`

export const Window = styled.div`
  position: fixed;
  top: calc(15% / 2);
  left: calc(15% / 2);
  height: 85%;
  width: 85%;
  overflow: hidden;
  opacity: 1;
  z-index: 999;
  text-align: right;
`
export const CloseIcon = styled.img`
  width: 17px;
  padding-bottom: 5px;
`

export const InnerWindow = styled.div`
  background: white;
  height: calc(100% - 30px);
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

`


export const LeftRightBorders = styled.div`
  width: 87%;
  border-left: solid 1px ${silver_polish};
  border-right: solid 1px ${silver_polish};
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const TopBottomBorders = styled.div`
  height: 100%;
  width: 100%;

  @media(min-width: ${desktopBreakpoint}px) {
    border-top: solid 1px ${silver_polish};
    border-bottom: solid 1px ${silver_polish};
    height: 80%;
  }

  `

export const Content = styled.div`
  text-align: left;
  padding: 20px;
  max-height: calc(100% - 40px);
  overflow-y: scroll;  

  @media(min-width: ${tabletBreakpoint}px) {
    padding: 40px;
    max-height: calc(100% - 80px);

  }

`

export const Tile = styled.div`
  width: 100%;
  height: 100%;
  background: ${off_black};
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
  padding-bottom: 25px;

  @media(min-width: ${tabletBreakpoint}px) {
    max-width: 235px;
    padding: 40px;
    margin: auto;
  }

`

export const Circle = styled.div`
  border-radius: 50%;
  background: ${jet};
  height: 145px;
  width: 145px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 35px;
  margin-bottom: 25px;

  @media(min-width: ${tabletBreakpoint}px) {
    height: 185px;
    width: 185px;
  }

`

type LogoProps = {
  icon?: string
}

export const Logo = styled.img<LogoProps>`
  height: ${({ icon }) => {
    switch(icon) {
      case "nike":
        return "28px";
      case "slalom": 
        return "50px";
      default: 
        return "70px";
    }
  }}};

  @media(min-width: ${tabletBreakpoint}px) {
    height: ${({ icon }) => {
      switch(icon) {
        case "nike":
          return "38px";
        case "slalom": 
          return "60px";
        default: 
          return "90px";
      }
    }}};  
  }

`

export const Overview = styled.div`
  display: flex;
  flex-direction: column;
  row-gap: 20px;

  @media(min-width: ${tabletBreakpoint}px) {
    flex-direction: row;
    column-gap: 20px;
    padding-top: 75px;
  }

`

export const TileLabel = styled.div`
  font-family: Gilroy-Bold;
  font-size: 14px;

  @media(min-width: ${tabletBreakpoint}px) {
    font-size: 17px;
    padding-bottom: 15px;
  }

`

export const TileInfo = styled.span`
font-family: Gilroy-Regular;
font-size: 14px;

@media(min-width: ${tabletBreakpoint}px) {
  font-size: 13px;
}
`

export const OrgTeam = styled.div`
  display: flex;
  flex-direction: column;

  @media(min-width: ${tabletBreakpoint}px) {
    flex-direction: row;
  }
`