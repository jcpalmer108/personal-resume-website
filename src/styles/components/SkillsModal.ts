import styled from "styled-components";
import {
  desktopBreakpoint,
  lively_coral,
  redstone,
  silver_polish,
  tabletBreakpoint,
} from "../variables";

export const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-wrap: wrap;
  text-align: left;
  justify-content: left;
  padding: 30px 0px;
  height: fit-content;

  @media (min-width: ${desktopBreakpoint}px) {
    justify-content: center;
    align-items: center;
    overflow-y: hidden;
    padding-top: 0px;
  }
`;

export const Section = styled.div`
  width: 100%;

  @media (min-width: ${tabletBreakpoint}px) {
    width: 50%;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    width: calc(33.3% - 0.5px);
    height: 286px;
    border-bottom: solid 1px ${silver_polish};
    border-right: solid 1px ${silver_polish};
    overflow-y: scroll;

    &:nth-child(4),
    &:nth-child(5),
    &:nth-child(6) {
      border-bottom: none;
    }

    &:nth-child(3),
    &:nth-child(6) {
      border-right: none;
    }
  }
`;

export const CallToAction = styled(Section)`
  background: blue;
  background: ${redstone};
  display: none;

  @media (min-width: ${desktopBreakpoint}px) {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
`;
export const Circle = styled.div`
  height: 90px;
  width: 90px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${lively_coral};
`;

export const RocketIcon = styled.img`
  height: 25px;
`;

export const Details = styled.div`
  color: white;
  font-family: Gilroy-Regular;
  font-size: 18px;
  weight: 600;
  padding-top: 15px;
  padding-bottom: 30px;
  max-width: 176px;
  text-align: center;
`;

export const Action = styled.div`
  background: white;
  color: ${redstone};
  font-family: Gilroy-Medium;
  font-size: 10px;
  padding: 20px;
  max-width: 176px;
`;

export const Subheader = styled.div`
  font-family: Gilroy-Bold;
  font-size: 22px;
  padding: 20px 30px 15px 50px;

  @media (min-width: ${tabletBreakpoint}px) {
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding: 23px 30px 15px 50px;
  }
`;

export const Line = styled.div`
  display: flex;
  column-gap: 15px;
  padding: 7px 50px;

  &:last-child {
    padding-bottom: calc(35px - 7px);
  }
`;

export const Icon = styled.img`
  height: 20px;
`;

export const Subtitle = styled.div`
  font-family: Gilroy-Regular;
  font-size: 11px;
  color: ${silver_polish};
  padding-top: 3px;
`;

export const Info = styled.div`
  font-family: Gilroy-Regular;
  font-size: 18px;

  @media (min-width: ${desktopBreakpoint}px) {
    font-size: 16px;
  }
`;

export const SkillInfo = styled.div`
  display: flex;
  flex-direction: column;
`;
