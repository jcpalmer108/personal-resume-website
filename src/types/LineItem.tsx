export type LineItemProps = {
  icon: string;
  skill: string;
  subtitle?: string;
};
