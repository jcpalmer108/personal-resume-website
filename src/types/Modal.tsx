import { ReactNode } from "react";

export type ModalProps = {
  closeModal: Function,
  children: ReactNode
}
