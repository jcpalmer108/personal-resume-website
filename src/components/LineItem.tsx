import { LineItemProps } from "../types/LineItem";
import {
  Icon,
  Info,
  Line,
  SkillInfo,
  Subtitle,
} from "../styles/components/LineItem";

export default function LineItem({ icon, skill, subtitle }: LineItemProps) {
  return (
    <Line>
      <Icon src={require("../assets/images/skills/modal/" + icon + ".svg")} />
      <SkillInfo>
        <Info>{skill}</Info>
        {subtitle && <Subtitle>{subtitle}</Subtitle>}
      </SkillInfo>
    </Line>
  );
}
