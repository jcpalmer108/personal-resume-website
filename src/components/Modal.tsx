import closeIcon from "../assets/images/icons/close.svg";
import {
  BackgroundOverlay,
  CloseIcon,
  InnerWindow,
  LeftRightBorders,
  TopBottomBorders,
  Window,
} from "../styles/components/Modal";
import { ModalProps } from "../types/Modal";

export default function Modal({ closeModal, children }: ModalProps) {
  return (
    <>
      <BackgroundOverlay data-testid="Overlay" onClick={() => closeModal()} />
      <Window data-testid="Modal">
        <CloseIcon
          src={closeIcon}
          alt="close"
          data-testid="CloseIcon"
          onClick={() => closeModal()}
        />
        <InnerWindow>
          <LeftRightBorders>
            <TopBottomBorders data-testid="ChildrenWrapper">
              {children}
            </TopBottomBorders>
          </LeftRightBorders>
        </InnerWindow>
      </Window>
    </>
  );
}
