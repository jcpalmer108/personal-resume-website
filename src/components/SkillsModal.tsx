import rocketIcon from "../assets/images/icons/rocket.svg";
import {
  Action,
  CallToAction,
  Circle,
  Details,
  RocketIcon,
  Section,
  Subheader,
  Wrapper,
} from "../styles/components/SkillsModal";
import { Skill } from "../types/content";
import { SkillsModalProps } from "../types/SkillsModal";
import Modal from "./Modal";
import LineItem from "./LineItem";
import downloadResume from "../utils/downloadResume";

export default function SkillsModal({
  closeModal,
  content,
  categories,
}: SkillsModalProps) {
  if (!content || !categories) {
    return null;
  }

  return (
    <Modal closeModal={closeModal}>
      <Wrapper data-testid="SkillsModal">
        {categories?.map((category, index) => (
          <Section key={`Section ${index + 1}`} data-testid="Section">
            <Subheader data-testid="Subheader">{category.label}</Subheader>
            <div>
              {content
                ?.filter((skill: Skill) => skill.category === category.key)
                .map((skill: Skill) => (
                  <LineItem
                    key={skill.key}
                    icon={skill.key}
                    skill={skill.label}
                    subtitle={skill.subtitle}
                  />
                ))}
            </div>
          </Section>
        ))}
        <CallToAction data-testid="CallToAction">
          <Circle>
            <RocketIcon src={rocketIcon} alt="rocket" />
          </Circle>
          <Details>There's more where that came from.</Details>
          <Action data-testid="ActionButton" onClick={(e) => downloadResume(e)}>
            {"Download my resume".toUpperCase()}
          </Action>
        </CallToAction>
      </Wrapper>
    </Modal>
  );
}
