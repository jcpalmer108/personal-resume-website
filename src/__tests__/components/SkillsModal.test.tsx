import { LineItemProps } from "@/types/LineItem";
import { fireEvent, render, screen } from "@testing-library/react";
import "jest-styled-components";
import SkillsModal from "../../components/SkillsModal";
import { ModalProps } from "../../types/Modal";

// mock helpers
const mockDownloadResume = jest.fn();
jest.mock("../../utils/downloadResume", () => () => mockDownloadResume());

// mock components
const mockModal = jest.fn();
jest.mock(
  "../../components/Modal",
  () =>
    ({ closeModal, children }: ModalProps) => {
      mockModal(closeModal, children);
      return <div>{children}</div>;
    }
);

const mockLineItems = jest.fn();
jest.mock(
  "../../components/LineItem",
  () =>
    ({ icon, skill, subtitle }: LineItemProps) => {
      mockLineItems(icon, skill, subtitle);
      return <div />;
    }
);

// params for component
const params = {
  required: {
    closeModal: jest.fn(),
  },
  optional: {
    content: [
      {
        key: "test",
        label: "One",
        category: "frontend",
      },
      {
        key: "test",
        label: "Three",
        category: "frontend",
      },
      {
        key: "test",
        label: "Two",
        category: "backend",
      },
    ],
    categories: [
      {
        key: "frontend",
        label: "Frontend",
      },
      {
        key: "backend",
        label: "Backend",
      },
    ],
  },
};

describe("SkillsModal", () => {
  test("renders if required and optional params are passed in", () => {
    // given
    const { required, optional } = params;
    render(
      <SkillsModal
        closeModal={required.closeModal}
        content={optional.content}
        categories={optional.categories}
      />
    );

    // then
    expect(screen.getByTestId("SkillsModal")).toMatchSnapshot();
    expect(screen.getByTestId("SkillsModal").childNodes).toHaveLength(
      optional.content.length
    );
    expect(screen.getAllByTestId("Subheader")).toHaveLength(
      optional.categories.length
    );
    optional.categories.forEach((category, index) => {
      const categoryNode = screen.getByTestId("SkillsModal").childNodes[index];
      expect(categoryNode.childNodes[0]).toHaveTextContent(category.label);
      expect(categoryNode.childNodes[1].childNodes).toHaveLength(
        optional.content.filter((skills) => skills.category === category.key)
          .length
      );
    });
    expect(screen.getByTestId("CallToAction")).toBeTruthy();
    expect(screen.getByTestId("CallToAction")).toHaveTextContent(
      "There's more where that came from."
    );
    expect(screen.getByTestId("ActionButton")).toHaveTextContent(
      "Download my resume".toUpperCase()
    );
  });

  test("does not render if content and categories are not passed in", () => {
    // given
    const { required } = params;
    render(<SkillsModal closeModal={required.closeModal} />);

    // then
    expect(screen.queryByTestId("SkillsModal")).toBeFalsy();
  });

  test("does not render if categories is not passed in", () => {
    // given
    const { required, optional } = params;
    render(
      <SkillsModal
        closeModal={required.closeModal}
        content={optional.content}
      />
    );

    // then
    expect(screen.queryByTestId("SkillsModal")).toBeFalsy();
  });

  test("does not render if content is not passed in", () => {
    // given
    const { required, optional } = params;
    render(
      <SkillsModal
        closeModal={required.closeModal}
        categories={optional.categories}
      />
    );

    // then
    expect(screen.queryByTestId("SkillsModal")).toBeFalsy();
  });

  test("executes form download on ActionButton click", () => {
    // given
    const { required, optional } = params;
    render(
      <SkillsModal
        closeModal={required.closeModal}
        content={optional.content}
        categories={optional.categories}
      />
    );

    // when
    fireEvent.click(screen.getByTestId("ActionButton"));

    // then
    expect(mockDownloadResume).toHaveBeenCalled();
  });
});
