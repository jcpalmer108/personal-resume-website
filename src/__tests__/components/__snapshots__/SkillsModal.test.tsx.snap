// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsModal renders if required and optional params are passed in 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  text-align: left;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  padding: 30px 0px;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c1 {
  width: 100%;
}

.c3 {
  background: blue;
  background: #e56b6f;
  display: none;
}

.c4 {
  height: 90px;
  width: 90px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #E7777A;
}

.c5 {
  height: 25px;
}

.c6 {
  color: white;
  font-family: Gilroy-Regular;
  font-size: 18px;
  weight: 600;
  padding-top: 15px;
  padding-bottom: 30px;
  max-width: 176px;
  text-align: center;
}

.c7 {
  background: white;
  color: #e56b6f;
  font-family: Gilroy-Medium;
  font-size: 10px;
  padding: 20px;
  max-width: 176px;
}

.c2 {
  font-family: Gilroy-Bold;
  font-size: 22px;
  padding: 20px 30px 15px 50px;
}

@media (min-width:1024px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    overflow-y: hidden;
    padding-top: 0px;
  }
}

@media (min-width:720px) {
  .c1 {
    width: 50%;
  }
}

@media (min-width:1024px) {
  .c1 {
    width: calc(33.3% - 0.5px);
    height: 286px;
    border-bottom: solid 1px #c6c6c6;
    border-right: solid 1px #c6c6c6;
    overflow-y: scroll;
  }

  .c1:nth-child(4),
  .c1:nth-child(5),
  .c1:nth-child(6) {
    border-bottom: none;
  }

  .c1:nth-child(3),
  .c1:nth-child(6) {
    border-right: none;
  }
}

@media (min-width:1024px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:720px) {

}

@media (min-width:1024px) {
  .c2 {
    padding: 23px 30px 15px 50px;
  }
}

<div
  class="c0"
  data-testid="SkillsModal"
>
  <div
    class="c1"
    data-testid="Section"
  >
    <div
      class="c2"
      data-testid="Subheader"
    >
      Frontend
    </div>
    <div>
      <div />
      <div />
    </div>
  </div>
  <div
    class="c1"
    data-testid="Section"
  >
    <div
      class="c2"
      data-testid="Subheader"
    >
      Backend
    </div>
    <div>
      <div />
    </div>
  </div>
  <div
    class="c1 c3"
    data-testid="CallToAction"
  >
    <div
      class="c4"
    >
      <img
        alt="rocket"
        class="c5"
        src="rocket.svg"
      />
    </div>
    <div
      class="c6"
    >
      There's more where that came from.
    </div>
    <div
      class="c7"
      data-testid="ActionButton"
    >
      DOWNLOAD MY RESUME
    </div>
  </div>
</div>
`;
